server/src/main/java/org/elasticsearch/index/mapper/DotExpandingXContentParser.java

```diff
@@ -232,7 +269,7 @@ public Token currentToken() {
     @Override
     public String currentName() throws IOException {
         if (state == State.PARSING_ORIGINAL_CONTENT) {
-            assert expandedTokens == subPaths.length * 2 - 1;
+            // assert expandedTokens == subPaths.length * 2 - 1;
```

Comment:
If you're replacing `subPaths` when we collapse, doesn't this assertion still hold?

Reply:
yes leftover, I commented it out to try a different approach that did not require replacing the array, but it turned out not to be easy at all, so I reverted that. Will reinstate the assertion.